app/controllers/amp_articles_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AmpArticlesController < ActionController::Base
+  def show
+    if (@article = Article.find_by_permalink(params[:from])).present?
+      if @article.supports_amp || params[:no_redirect]
+        respond_to do |format|
+          format.html do
+            render 'articles/amp/show'
+          end
+        end
+      else
+        redirect_to article_url(@article.permalink)
+      end
+    else
+      render 'errors/404', status: 404
+    end
+  end
+
+  def article_body(article)
+    body = article.html(:body)
+    page = Nokogiri::HTML.fragment(body)
+    page = process_img_tag(page)
+    page = process_iframe_tag(page)
+    page
+  end
+  helper_method :article_body
```

Comment:
@shaley91 look [here](https://github.com/moneyadviceservice/cms/blob/master/app/models/table_captioner.rb) for some inspiration on how this could be implemented.